compiler/src/dotty/tools/dotc/printing/Formatting.scala

```diff
@@ -307,18 +316,22 @@ object Formatting {
   }
 
   private def errorMessageCtx(using Context): Context =
-    ctx.property(MessageLimiter) match
-      case Some(_: ErrorMessageLimiter) => ctx
-      case _ => ctx.fresh.setProperty(MessageLimiter, ErrorMessageLimiter())
+    ctx.withProperty(MessageLimiter)(ErrorMessageLimiter()).withPrinter(new ErrorMessagePrinter(_))
```

Comment:
For my taste that's overengineered. I'd avoid introducing a different withProperty with curried parameters and would stick more or less to the previous logic. I.e. test for `MessageLimiter` property, install if it is missing.